rust/core/src/ffi/types.rs

```diff
@@ -17,4 +17,622 @@
+    pub(crate) StatementNew: Option<methods::FuncStatementNew>,
+    pub(crate) StatementPrepare: Option<methods::FuncStatementPrepare>,
+    pub(crate) StatementRelease: Option<methods::FuncStatementRelease>,
+    pub(crate) StatementSetOption: Option<methods::FuncStatementSetOption>,
+    pub(crate) StatementSetSqlQuery: Option<methods::FuncStatementSetSqlQuery>,
+    pub(crate) StatementSetSubstraitPlan: Option<methods::FuncStatementSetSubstraitPlan>,
+    pub(crate) ErrorGetDetailCount: Option<methods::FuncErrorGetDetailCount>,
+    pub(crate) ErrorGetDetail: Option<methods::FuncErrorGetDetail>,
+    pub(crate) ErrorFromArrayStream: Option<methods::FuncErrorFromArrayStream>,
+    pub(crate) DatabaseGetOption: Option<methods::FuncDatabaseGetOption>,
+    pub(crate) DatabaseGetOptionBytes: Option<methods::FuncDatabaseGetOptionBytes>,
+    pub(crate) DatabaseGetOptionDouble: Option<methods::FuncDatabaseGetOptionDouble>,
+    pub(crate) DatabaseGetOptionInt: Option<methods::FuncDatabaseGetOptionInt>,
+    pub(crate) DatabaseSetOptionBytes: Option<methods::FuncDatabaseSetOptionBytes>,
+    pub(crate) DatabaseSetOptionDouble: Option<methods::FuncDatabaseSetOptionDouble>,
+    pub(crate) DatabaseSetOptionInt: Option<methods::FuncDatabaseSetOptionInt>,
+    pub(crate) ConnectionCancel: Option<methods::FuncConnectionCancel>,
+    pub(crate) ConnectionGetOption: Option<methods::FuncConnectionGetOption>,
+    pub(crate) ConnectionGetOptionBytes: Option<methods::FuncConnectionGetOptionBytes>,
+    pub(crate) ConnectionGetOptionDouble: Option<methods::FuncConnectionGetOptionDouble>,
+    pub(crate) ConnectionGetOptionInt: Option<methods::FuncConnectionGetOptionInt>,
+    pub(crate) ConnectionGetStatistics: Option<methods::FuncConnectionGetStatistics>,
+    pub(crate) ConnectionGetStatisticNames: Option<methods::FuncConnectionGetStatisticNames>,
+    pub(crate) ConnectionSetOptionBytes: Option<methods::FuncConnectionSetOptionBytes>,
+    pub(crate) ConnectionSetOptionDouble: Option<methods::FuncConnectionSetOptionDouble>,
+    pub(crate) ConnectionSetOptionInt: Option<methods::FuncConnectionSetOptionInt>,
+    pub(crate) StatementCancel: Option<methods::FuncStatementCancel>,
+    pub(crate) StatementExecuteSchema: Option<methods::FuncStatementExecuteSchema>,
+    pub(crate) StatementGetOption: Option<methods::FuncStatementGetOption>,
+    pub(crate) StatementGetOptionBytes: Option<methods::FuncStatementGetOptionBytes>,
+    pub(crate) StatementGetOptionDouble: Option<methods::FuncStatementGetOptionDouble>,
+    pub(crate) StatementGetOptionInt: Option<methods::FuncStatementGetOptionInt>,
+    pub(crate) StatementSetOptionBytes: Option<methods::FuncStatementSetOptionBytes>,
+    pub(crate) StatementSetOptionDouble: Option<methods::FuncStatementSetOptionDouble>,
+    pub(crate) StatementSetOptionInt: Option<methods::FuncStatementSetOptionInt>,
+}
+
+unsafe impl Send for FFI_AdbcDriver {}
+unsafe impl Sync for FFI_AdbcDriver {}
```

Comment:
Can you also add a safety note here?

Reply:
Same reason as in https://github.com/apache/arrow-adbc/pull/1742#discussion_r1575281649